orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py

```diff
@@ -241,39 +241,42 @@ def forward(self):
         All other methods are internal"""
         pass
 
-    def _build_graph(self):
+    def _build_graph(self, training):
         if self._use_static_shape:
             self._graph_builder.build(self._input_info.shape)
         else:
             self._graph_builder.build()
 
-        self._onnx_models.optimized_model = onnx.load_model_from_string(self._graph_builder.get_model())
-
-        self._onnx_models.optimized_pre_grad_model = onnx.load_model_from_string(
-            self._graph_builder.get_inference_optimized_model()
-        )
+        if training:
+            self._onnx_models.optimized_model = onnx.load_model_from_string(self._graph_builder.get_gradient_model())
+            self._onnx_models.optimized_pre_grad_model = onnx.load_model_from_string(
```

Comment:
why do we need the optimized_pre_grad_model? In the training mode once the gradient graph is ready will we ever use this? We can free up mem if we dont need it

Reply:
I guess it's useful for debugging when we need to save these graphs to disk for comparison. Normally mem is not an issue for ORTModule since all weights are graph inputs so the graph is relatively small most of the time.